addons/microsoft_calendar/models/calendar.py

```diff
@@ -82,7 +82,7 @@ def create(self, vals_list):
         if self._check_microsoft_sync_status() and not notify_context and recurrency_in_batch:
             self._forbid_recurrence_creation()
 
-        for vals in vals_list:
+        for vals in [v for v, b in zip(vals_list, self._check_organizer_validation_conditions(vals_list)) if b]:
```

Comment:
More explicit names are better, also might be easier to filter values directly instead of zipping with booleans
```suggestion
        vals_check_organizer = self._check_organizer_validation_conditions(vals_list)
        for vals in [vals for vals, check_organizer in zip(vals_list, vals_check_organizer) if check_organizer]:
```